transformer_engine/pytorch/fp8.py

```diff
@@ -26,6 +27,29 @@
 _amax_forward_global_reduce_func = None
 _buffer_delete_key_fwd = None
 _buffer_delete_key_bwd = None
+_is_fp8_available = None
+
+
+def _check_fp8_support() -> Tuple[bool, str]:
+    """Return if fp8 support is available"""
+    if get_device_compute_capability() >= 9.0: # hopper and above
+        return True, ""
+    if get_device_compute_capability() < 8.9: # pre-ada
+        return False, "Device compute capability 8.9 or higher required for FP8 execution."
+    if tex.get_cublasLt_version() < 120103:
+        return False, "CublasLt version 12.1.3.x or higher required for FP8 execution on Ada."
+    if float(torch.version.cuda) < 12.1:
+        return False, "Cuda version 12.1 or higher required for FP8 execution on Ada."
+    return True, ""
+
+
+def is_fp8_available() -> Tuple[bool, str]:
+    """Return if fp8 support is available"""
+    global _is_fp8_available
+    reason_for_no_fp8 = ""
+    if _is_fp8_available is None:
+        _is_fp8_available, reason_for_no_fp8 = _check_fp8_support()
+    return _is_fp8_available, reason_for_no_fp8
```

Comment:
I don't like that we only provide `reason_for_no_fp8` the first time we call this function and would prefer if we cached it like `_is_fp8_available`. Alternatively, could we skip caching entirely and replace this with `_check_fp8_support`? This would be helpful when running on a machine with multiple GPUs.